snuba/query/snql/parser.py

```diff
@@ -394,20 +586,85 @@ def visit_function_call(
         self,
         node: Node,
         visited_children: Tuple[
-            str, Any, List[Expression], Any, Union[Node, List[Expression]]
+            str,
+            Any,
+            List[Expression],
+            Any,
+            Union[Node, List[Expression]],
+            Union[Node, List[Any]],
         ],
     ) -> Expression:
-        return visit_function_call(node, visited_children)
+        name, _, params1, _, params2, alias = visited_children
+        if isinstance(alias, Node) or len(alias) == 0:
+            alias = None
+        else:
+            _, _, _, alias = alias
+            alias = alias.text
+
+        param_list1 = tuple(params1)
+        internal_f = FunctionCall(alias, name, param_list1)
```

Comment:
If this is a curried function the alias should be applied to the CurriedFunctionCall, not to the internal function.